pkg/scalers/kafka_scaler.go

```diff
@@ -118,6 +155,22 @@ func getKafkaClients(metadata kafkaMetadata) (sarama.Client, sarama.ClusterAdmin
 	config := sarama.NewConfig()
 	config.Version = sarama.V1_0_0_0
 
+	if ok := metadata.authMode == kafkaAuthModeForSaslPlaintext || metadata.authMode == kafkaAuthModeForSaslScramSha256 || metadata.authMode == kafkaAuthModeForSaslScramSha512; ok {
+		config.Net.SASL.Enable = true
+		config.Net.SASL.User = metadata.username
+		config.Net.SASL.Password = metadata.passwd
+	}
+
+	if metadata.authMode == kafkaAuthModeForSaslScramSha256 {
```

Comment:
I’m not familiar enough with how Kafka does SASL, but does the user need to reference a cert or anything? Or does it just need a cert that it appears the scaler can generate and should work.  Just curious if we need to document how to reference your own certs if required (or provide hooks to do so)

Reply:
hi @jeffhollan . you are right.  i add support for sasl_ssl.